app/services/rag_wise/discussion_extraction_service.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+class RagWise::DiscussionExtractionService
+  def initialize(course, topic, posts)
+    @course = course
+    @topic = topic
+    @posts = Course::Discussion::Post.where(id: posts.pluck(:id))
+  end
+
+  def call
+    {
+      topic_title: sanitized_topic_title,
+      discussion: formatted_discussion
+    }
+  end
+
+  private
+
+  def sanitized_topic_title
+    sanitize_text(@topic[:title])
+  end
+
+  def formatted_discussion
+    @posts.filter_map do |post|
+      sanitized_text = sanitize_text(post[:text])
+
+      {
+        role: post_creator_role(@course, post),
+        text: sanitized_text
```

Comment:
Might as well skip the single-use variable in L24 and assign directly here:  
`text: sanitize_text(post[:text])`

Reply:
True. I will edit it